src/lib/helpers/ripple.ts

```diff
@@ -0,0 +1,99 @@
+interface RippleOptions {
+  center?: boolean;
+  color?: string; // CSS color
+  duration?: number; // In ms
+}
+
+const triggerEvents: any[] = ["pointerdown", "touchstart", "keydown"];
+const cancelEvents = ["mouseleave", "dragleave", "touchmove", "touchcancel", "pointerup", "keyup"];
+
+export function ripple(el: HTMLElement, options: RippleOptions = {}) {
+  function addClasses(center?: boolean) {
+    let shouldBeCentered = center || options.center;
+
+    !el.classList.contains("q-ripple--effect") && el.classList.add("q-ripple--effect");
+
+    !shouldBeCentered &&
+      el.classList.contains("q-ripple--center") &&
+      el.classList.remove("q-ripple--center");
+
+    shouldBeCentered && el.classList.add("q-ripple--center");
+  }
+
+  function setOptions(options: RippleOptions) {
+    if (options.duration && options.duration <= 0) {
+      options.duration = undefined;
+    }
+
+    options.color && el.style.setProperty("--ripple-color", options.color);
+
+    options.duration && el.style.setProperty("--ripple-duration", `${options.duration}ms`);
+  }
+
+  addClasses();
+  setOptions(options);
+
+  function createRipple(e: PointerEvent | KeyboardEvent, center?: boolean) {
+    if (el.hasAttribute("aria-disabled")) return;
+
+    if (e instanceof KeyboardEvent) {
+      if ((e.code !== "Enter" && e.code !== "Space") || e.repeat) {
```

Comment:
Optional: You could use `!['Enter', 'Space'].includes(e.code)` instead, which is more readable but maybe not quite as performant. It's not important though.